src/functional/ffront/type_info.py

```diff
@@ -277,6 +277,8 @@ def is_concretizable(symbol_type: ct.SymbolType, to_type: ct.SymbolType) -> bool
     True
 
     """
+    if isinstance(symbol_type, ct.FunctionType):
+        raise NotImplementedError("`FunctionType` is not concretizable")
```

Comment:
What's this change about? Shouldn't this case be handled in the `is_concrete()` case?

Reply:
In a prior version of this PR @nfarabullini used something along this lines
```suggestion
ct.FunctionType(args=[ct.DeferredSymbolType(constraint=...)], ...)
```
in `fbuiltins.py` to restrict the arguments of the number type constructors. Currently this will wrongly return `False` here. `is_concrete` also does not handle the case correctly and should consequently also throw a `NotImplementedError`. However I would argue that the check is still valid here as after adding support for these cases in `is_concrete` this function would again return `False`, which it shouldn't.